### PR TITLE
Set starting XP for starting heroes

### PR DESCRIPTION
**Pull request by eos428**
Solves #1316.

Latest behavior:
- Starting heroes receive xp from 40 to 90
- Tavern heroes still start with 0 xp (Edit: Now they also receive starting xp)

### CODE REVIEW

**Comment by LeHerosInconnu** [comment]
Hi @eos428,

> Solves #1316.
> 
> Latest behavior:
> 
> * Starting heroes receive xp from 40 to 90
> * Tavern heroes still start with 0 xp

A small clarification on this subject.
When I wrote "all starting heroes" that meant "all the heroes of the hero pool at the start of the scenario (all possible heroes)", that is to say all the heroes who have not yet been recruited and who have not yet gained experience points.

In fact, in the original game, all the heroes have some experience, the ones that are available in the tavern as well, and all the heroes from the hero pool (all possible heroes, those that will be available later in the tavern).
Experience points are certainly granted to all possible heroes at the start of the scenario and the values are then saved for all save files.

**Comment by eos428** [comment]
Alright, now the tavern heroes should get starting xp too :D

**Comment by LeHerosInconnu** [comment]
@eos428.

> Alright, now the tavern heroes should get starting xp too :D

Nice, thanks for the changes. :)

File: src/fheroes2/heroes/heroes.h [hunk 1]
```diff
@@ -173,6 +173,7 @@ class Heroes : public HeroBase, public ColorBase
     int GetColor( void ) const;
     int GetType( void ) const;
     int GetControl( void ) const;
+    u32 GetStartingXp( void ) const;
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes.h:176]: Please use C++ type `uint32_t` here and in the implemention part as well. **u32** is a part of the old project which we try to remove. Also please do not put `void` in input parameters, it is a redundant input parameter (also the old code, eh...)

Comment (ihhub) [reply src/fheroes2/heroes/heroes.h:176]: Please make this method `static` under `private` access modifier.

Comment (eos428) [reply src/fheroes2/heroes/heroes.h]: Done :)

**Comment by ihhub** [comment]
@eos428 . excellent work! Thanks a lot for fixing this issue!

**Pull request merged**